apps/desktop/src/app/sync/sync.service.ts

```diff
@@ -91,55 +106,91 @@ export class SyncService {
+    this.authenticateAllPendingConnectors()
+      .pipe(
+        concatMapTo(
+          combineLatest(
+            this._queue
+              .filter((sync) => !sync.started && !sync.completed)
+              .map((sync) => {
+                return combineLatest([
+                  this.getSource(sync.source).pipe(take(1)),
+                  this.getDestination(sync.destination.id).pipe(take(1)),
+                ]).pipe(
+                  switchMap(([sourceInstance, destinationInstance]) => {
+                    sync.started = true;
+                    // TODO: go 6 by 6 maximum
+                    return forkJoin(
+                      sync.files
+                        .filter((f) => f.status === FileStatus.PENDING)
+                        .map((f) =>
+                          sourceInstance.download(f).pipe(
+                            concatMap((blob) => {
+                              if (sync.destination.id === 'nucliacloud') {
+                                return destinationInstance.upload(f.title, blob, sync.destination.params).pipe(
+                                  tap(() => {
+                                    f.status = FileStatus.UPLOADED;
+                                    this.onQueueUpdate();
+                                  }),
+                                );
+                              } else {
+                                return this.sdk.nuclia.db.upload(new File([blob], f.title)).pipe(
+                                  tap(() => {
+                                    f.status = FileStatus.PROCESSED;
+                                    this.onQueueUpdate();
+                                  }),
+                                );
+                              }
                             }),
-                          );
-                        }
-                      }),
-                      take(1),
-                    ),
-                  ),
-              );
-            }),
-            tap(() => {
-              sync.completed = true;
-              this.onQueueUpdate();
-            }),
+                            take(1),
+                          ),
+                        ),
+                    );
+                  }),
+                  tap(() => {
+                    sync.completed = true;
+                    this.onQueueUpdate();
+                  }),
+                );
+              }),
           ),
         ),
-    ).subscribe();
+      )
+      .subscribe();
+  }
+
+  authenticateAllPendingConnectors(): Observable<void> {
+    const sources = this._queue
+      .filter((sync) => !sync.started && !sync.completed)
+      .reduce((acc, sync) => {
+        if (!acc[sync.source]) {
+          acc[sync.source] = this.getSource(sync.source).pipe(
+            take(1),
+            concatMap((source) => {
+              if (source.hasServerSideAuth) {
+                // TODO: we cannot redirect to several third party login pages at the same time
```

Comment:
I don't understand the meaning of this todo, it seems like you're describing something you cannot do instead of what we should try to do 🤔.